transports/quic/src/transport.rs

```diff
@@ -172,6 +173,22 @@ impl<P: Provider> GenTransport<P> {
             }
         }
     }
+
+    fn create_socket(&self, socket_addr: SocketAddr) -> io::Result<UdpSocket> {
+        let domain = if socket_addr.is_ipv4() {
+            Domain::IPV4
+        } else {
+            Domain::IPV6
+        };
+        let socket = Socket::new(domain, Type::DGRAM, Some(socket2::Protocol::UDP))?;
```

Comment:
```suggestion
        let socket = Socket::new(socket2::for_address(socket_addr), Type::DGRAM, Some(socket2::Protocol::UDP))?;
```

How about using [`for_addr`](https://docs.rs/socket2/latest/socket2/struct.Domain.html#method.for_address) here and in `libp2p-tcp`?

Reply:
right, makes total sense. Thanks Max updated :) I can submit the fix for `libp2p-tcp` on another PR.